src/uu/rm/src/rm.rs

```diff
@@ -370,6 +370,45 @@ fn prompt_file(path: &Path, is_dir: bool) -> bool {
     }
 }
 
+// copied from https://github.com/rust-lang/cargo/blob/2e4cfc2b7d43328b207879228a2ca7d427d188bb/src/cargo/util/paths.rs#L65-L90
+// both projects are MIT https://github.com/rust-lang/cargo/blob/master/LICENSE-MIT
+// for std impl progress see rfc https://github.com/rust-lang/rfcs/issues/2208
+// replace this once that lands
+pub fn normalize_path(path: &Path) -> PathBuf {
```

Comment:
Since some of the lines of this function were not covered by the test case, I see three options:

1. Ignore the checks (probably not desired)
2. Add extra tests to `rm`
3. Promote this into a core util function and write unit tests for it for coverage
4. do both 2. and 3.

Let me know what you think

Reply:
please, do 3 :)

Reply:
Hey @sylvestre,

I am trying to write unit tests for the now extracted function and noticed that there is no framework to actually test those

I started laying the foundation for that but came to a blocker where the tests I write can't actually detect if a dependency has a feature enabled, which is important as I can only run those tests if `uucore/fs` is enabled

My options now are:

1. putting it into core and skipping unit tests
2. feature gate the tests for the `rm` feature (means it can get out of sync)
3. You might have an idea how to better test `uucore`